### DISCUSSION TITLE
Support for star ratings

### DISCUSSION THREAD

**Discussion by madbrain76** [music-assistant/support#2461] [Feature requests and ideas]
Most music providers currently use a simple favorite or like/dislike system. There are a few that do not, however, and use a star ratings instead .
1. Plex
2. Apple music, with provider currently in beta
3. DLNA servers such as Jriver music center (currently not supported)
4. Potentially the file system provider, for file types that support that Metadata

I'm not sure what the support would look like. Star ratings could be exposed through the UI. Or it could be hidden, and there could be some mapping between the two systems, such as 5 stars being equivalent to favorites.

**Comment by klodeckl** [comment]
I would love to see star ratings in general for local music.

**Comment by hannashimona** [comment]
Would also love to see this!

**Comment by screenagerbe** [comment]
same here.
I'm using the stars to refer to the physical format of the album, if its in our collection(s)

**Comment by v-marinkov** [comment]
I have been labelling my collection in MusicBee and would love to be able to bring the star ratings to Music Assistant! Of course, to be full music management system, it would need to be able to write ratings back to the files as well.

**Comment by stefandewekker** [comment]
Would love to see support for (local file) ratings in MA. Ideally complemented with an optional setting to sync ratings to the files for use elsewhere (eq. portable audio players).

**Comment by Chippit** [comment]
Putting in my vote for this also. My music server (Navidrome), and other OpenSubsonic servers support both favouriting/starring music, as well as applying ratings. I use these independantly to produce automatic playlists, and being able to set the rating for the current song via automation in Home Assistant and via the Music Assistant frontend would be great.

**Comment by OzGav** [comment]
Update for the followers here. After discussions internally we dont plan to add this at this time. We don't see the use case that the existing favourite flag doesn't provide. Sorry.

**Reply by marcelveldt** [reply]
I think to add to this is the following:

- We can later revisit if there seems to be a huge interest
- We plan to make a good mapping between star rating --> favourite and vice versa

**Comment by gogglespisano** [comment]
In the music server I wrote, users can rate songs (stars) and create dynamic playlists based on star level or mutual user star level combinations for shared playlists.

The existing favorite flag doesn't come close to handling a star based use case.  You essentially get one playlist for the one star (favorite) level.

**Comment by Vergil365** [comment]
i am also interested in this, would be nice to see it implemented.

**Comment by sodfenrir** [comment]
I use it everyday with media monkey. The star rating is written in the tags and the songs are added, for exemple, into dynamic playlists. 
Use case: "Today, i just want my very favorites songs" -> 5 stars playlist. "Today, i just want to have some backgroud music" -> 3 stars playlist with hundreds of songs that i like and that i don't hear everyday.  I want to exclude the really bad songs that I have because other people have wrong tastes: exclude <=3 stars from "all the X genre music". 

"what music must I still rate?" -> no star.

**Reply by bchevreau** [reply]
This is the best and most simple explanation as to why star rating is so much more versatile than Favorites. It would even coexist imo if I had a choice.  My old entire iTunes library is built that way. 

Dynamic playlists based on star rating is incredibly useful, and when coupled with genre, can help you set the mood in 2 clicks what you'dneed hours of queue minmaxing for.

**Comment by Gamrin** [comment]
@sodfenrir Hey, media monkey mentioned.

I came from Media Monkey a long time ago, but now on Linux have moved to Quod Libet specifically because it has extensive support for any kind of the Rating (Stars). It looks like it's just a double in the ID3 tags, rendered as a fraction depending on what the user wants.

(0.0, 0.2, 0.4, 0.6, 0.8, 1) for a 5-star system
(0, 1) for a Favourite/one star system.

My song files use ID3 tags extensively. I'd love to see them on my music system.
In particular, I have tags for Ballroom Dancing. I'd love to be able to migrate away from Spotify 100%

I have a great interest in this feature being a part of Music Assistant, and I'm overjoyed to see it be the first issue in the list when I was looking for it.

**Comment by Noki410** [comment]
Just to add: I also would love the star rating addition!

**Comment by rcb-hp3000** [comment]
Have our home media collection on our NAS and have been using Windows Media Player to put star ratings on each track as to how much individual liked the track. With 20,000+ songs in the library a simple like/don't like doesn't give you a very meaningful mechanism... Have always used the auto-playlists to create a list of 3+, 4+ star tracks or 5-star tracks... Maybe the 2+ star tracks when I want a bigger variety. 
Also having to start from scratch and cycling through 20,000 tracks to find the ones I have tagged over the years is a daunting task. Would be really nice to have a means to import those ratings from Windows Media Player.

**Comment by marveljpm2** [comment]
Star ratings is so useful for smart playlists. Yes please

**Comment by hannashimona** [comment]
This continues, more than a year later, to be the primary issue preventing me from using music assistant as my primary player. I use star ratings as the criteria for creating smart playlists, making them the building block of my entire music categorization system. I know I'm not the only one!

**Reply by OzGav** [reply]
Music Assistant has its own smart playlist creation functionality.

**Reply by hannashimona** [reply]
Absolutely! I'd like to use it. And my methodology for using smart playlists involves segmenting my music based on its star rating -- I've invested in this infra for over a decade. Thousands of songs rated, dozens of smart playlist rules based on star ratings.

**Reply by hannashimona** [reply]
I think what's deficient for me about marking songs "liked" vs not, is that it doesn't allow me to differentiate between songs that I don't like and songs I haven't listened to / rated yet -- a workable rating system for me would need at least three possible values -- positive, negative, and neutral, to allow me to exclude "disliked" songs without excluding those I haven't listened to yet.

**Reply by Chippit** [reply]
Star ratings also provide a very elegant and understandable way to distinguish between 'loved' and 'liked' songs, which has always been one of my motivating use cases.

**Reply by jdou** [reply]
> I think what's deficient for me about marking songs "liked" vs not, is that it doesn't allow me to differentiate between songs that I don't like and songs I haven't listened to / rated yet -- a workable rating system for me would need at least three possible values -- positive, negative, and neutral, to allow me to exclude "disliked" songs without excluding those I haven't listened to yet.

Agree with this. I think the key is that songs fall into three categories:Like, Unrated and the more complex "dislike".  This would allow a smart playlist to play X% of liked tracks and throw in X% of unrated tracks to explore/resurface, whilst allowing a dislike to be use to ban the track from playing. This could be "skit" tracks common in 90's hip-hop or just a track you don't love.  

Why not just remove these tracks for the library? Two reasons, A) Keeping the album complete, and B) We may like listening to these songs in some contexts.

**Comment by schildbach** [comment]
While I agree that "0-5 star ratings" are perhaps a bit over the top for a personal music library, the one marker I'm always missing in MA is:

"I don't like this song" (would probably be equivalent to zero stars)

This info could be used to never queue it up in any form of autoplay, maybe skip it when playing an album, etc.

**Reply by Gamrin** [reply]
"I don't like this song" Would and could be encoded in the same byte.
Usually, a 5-star rating is just an equal divide of 255. If your push of the "dislike" button lowers the value of this by (Tunable for users (50)) every time you do, it'll give you agency and have the songs you like less wane. Meanwhile songs that you press like on often would get their value upped towards a ceiling.
